### PR TITLE
Soe 1149

### PR DESCRIPTION
**Pull request by kmakitan**
Created permissions tests for testing site owner and editor permissions.
Based off of this: https://sites-jumpstart.stanford.edu/jumpstart-user-guide/users/site-owner-and-editor-permissions
Google spreadsheet for fulfilled tests: https://docs.google.com/a/stanford.edu/spreadsheets/d/1ROAF1hwvSi__ozQWNNtDLOhv8XgKNn2fyFEpu_Rih0w/edit?usp=sharing
JIRA Ticket: https://stanfordits.atlassian.net/browse/SOE-1149

### CODE REVIEW

File: includes/features/SU-SWS/permissions_tests/clear_cache/clear_cache.feature [hunk 1]
```diff
@@ -0,0 +1,18 @@
+Feature: Ensure Site Owners and Editors can clear the site cache
+  In order to ensure that site owners and editors can clear the site cache
+  As a site owner or editor
+  I want to be able to clear the site cache
+
+    @api @dev @destructive @javascript
```
Comment (cjwest) [inline includes/features/SU-SWS/permissions_tests/clear_cache/clear_cache.feature:6]: We have determined that clearing the cache is _not_ destructive! Go ahead and replace @dev @destructive with @safe

File: includes/features/SU-SWS/permissions_tests/clear_cache/clear_cache.feature [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    Scenario: Clear site cache as site owner
+    Given I am logged in as a user with the "site owner" role
+    When I wait for the Site Actions drop down to appear
+    And I click "Clear Site Cache" in the "Admin Shortcuts" region
+    Then I should see "Site Cache Cleared" in the "Console" region
+
+    @api @dev @destructive @javascript
```
Comment (cjwest) [inline includes/features/SU-SWS/permissions_tests/clear_cache/clear_cache.feature:13]: replace @dev @destructive with @safe

**Comment by cjwest** [comment]
@kmakitan,
Because permissions can be very product and site specific, let's move the permissions tests into the JSE directory and copy/adjust them for each product on the user guide.

https://sites-jumpstart.stanford.edu/jumpstart-user-guide/users/site-owner-and-editor-permissions

File: includes/features/SU-SWS/permissions_tests/create_delete_affiliate_organization/create_delete_affiliate_organization.feature [hunk 3]
```diff
@@ -0,0 +1,99 @@
+Feature: Ensure Site Owners can create and delete affiliates
+  In order to ensure that users with permissions can create, edit and delete affiliates
+  As a user with site permissions
+  I want to be able create, edit and delete affiliates
+
+  @api @dev @destructive @javascript
+  Scenario: Create, edit, then delete affiliate as site owner
```
Comment (cjwest) [inline includes/features/SU-SWS/permissions_tests/create_delete_affiliate_organization/create_delete_affiliate_organization.feature:7]: Do we need 4 scenarios? What about just two? one for site owner and one for editor? If we need all four, let's expand the description in the scenario as to "why" not just "what."

File: includes/features/SU-SWS/permissions_tests/create_delete_event/create_delete_event.feature [hunk 4]
```diff
@@ -0,0 +1,98 @@
+Feature: Ensure Site Owners can create and delete events
+  In order to ensure that users with permissions can create, edit and delete events
+  As a user with site permissions
+  I want to be able create, edit and delete events
+
+  @api @dev @destructive @javascript
+  Scenario: Create, edit, then delete event as site owner
```
Comment (cjwest) [inline includes/features/SU-SWS/permissions_tests/create_delete_event/create_delete_event.feature]: Let's expand the description in the scenario as to "why" not just "what."

File: includes/features/SU-SWS/permissions_tests/create_delete_event/create_delete_event.feature [hunk 5]
```diff
@@ -0,0 +1,98 @@
+Feature: Ensure Site Owners can create and delete events
```
Comment (cjwest) [inline includes/features/SU-SWS/permissions_tests/create_delete_event/create_delete_event.feature:1]: Please update this, I see the editor is involved in some scenarios.

File: includes/features/SU-SWS/permissions_tests/create_delete_event/create_delete_event.feature [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+  Then I should see "Stanford Event Bar has been updated" in the "Console" region
+  And I should be on "events/bar"
+  And I am on "events/bar"
+  And I click "Edit" in the "Content Head" region
+  And I click on the element with css selector "#edit-delete"
+  And I click on the element with css selector "#edit-submit"
+  Then I should see the text "Stanford Event Bar has been deleted" in the "Console" region
+
+  @api @dev @destructive @javascript
+  Scenario: Create event as site owner, edit as editor, delete as site owner
```
Comment (cjwest) [inline includes/features/SU-SWS/permissions_tests/create_delete_event/create_delete_event.feature:29]: Please explain why we need this scenario.

File: includes/features/SU-SWS/permissions_tests/create_delete_event/create_delete_event.feature [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+  And I should be on "events/bar"
+  Given I am logged in as a user with the "site owner" role
+  And I am on "events/bar"
+  And I click "Edit" in the "Content Head" region
+  And I click on the element with css selector "#edit-delete"
+  And I click on the element with css selector "#edit-submit"
+  Then I should see the text "Stanford Event Bar has been deleted" in the "Console" region
+
+  @api @dev @destructive @javascript
+  Scenario: Create, edit event as editor, delete as a site owner
```
Comment (cjwest) [inline includes/features/SU-SWS/permissions_tests/create_delete_event/create_delete_event.feature:54]: Please explain why we need this scenario.

File: includes/features/SU-SWS/permissions_tests/create_delete_event/create_delete_event.feature [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+  And I should be on "events/bar"
+  Given I am logged in as a user with the "site owner" role
+  And I am on "events/bar"
+  And I click "Edit" in the "Content Head" region
+  And I click on the element with css selector "#edit-delete"
+  And I click on the element with css selector "#edit-submit"
+  Then I should see the text "Stanford Event Bar has been deleted" in the "Console" region
+
+  @api @dev @destructive @javascript
+  Scenario: Create event as editor, edit and delete as a site owner
```
Comment (cjwest) [inline includes/features/SU-SWS/permissions_tests/create_delete_event/create_delete_event.feature:77]: Please explain why we need this scenario.

File: includes/features/SU-SWS/permissions_tests/create_delete_event_series/create_delete_event_series.feature [hunk 9]
```diff
@@ -0,0 +1,51 @@
+Feature: Ensure Site Owners can create and delete event series, Editors can edit event series
```
Comment (cjwest) [inline includes/features/SU-SWS/permissions_tests/create_delete_event_series/create_delete_event_series.feature:1]: Caryl likes this description

File: includes/features/SU-SWS/permissions_tests/create_delete_gallery/create_delete_gallery.feature [hunk 10]
```diff
@@ -0,0 +1,53 @@
+Feature: Ensure Site Owners can create and delete galleries
```
Comment (cjwest) [inline includes/features/SU-SWS/permissions_tests/create_delete_gallery/create_delete_gallery.feature:1]: Explain role of editor too.

File: includes/features/SU-SWS/permissions_tests/create_delete_help_page/create_delete_help_page.feature [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+  As a site owner
+  I want to be able create, edit and delete the help page
+
+  @api @dev @destructive @javascript
+  Scenario: Create, edit, then delete a help page as site owner
+  Given I am logged in as a user with the "site owner" role
+  When I wait for the Site Actions drop down to appear
+  And I click "Manage Content" in the "Admin Shortcuts" region
+  And I click "Manage Help Pages" in the "Second sidebar" region
+  Then I should see "Create Person" in the "Branding" region
```
Comment (cjwest) [inline includes/features/SU-SWS/permissions_tests/create_delete_help_page/create_delete_help_page.feature:12]: This looks like tests for person not help pages.

File: products/jse_sub/features/jse_site_owner.feature [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+    When I click "Show Add/Edit Image"
+    And I wait 2 seconds
+    Then I should see "Source Info" in the "Content Body" region
+    Then I fill in "edit-title" with "Foo"
+    And I press the "Save" button
+    Then I should see "Stanford Page Foo has been created" in the "Console" region
+    And I should be on "foo"
+
+  @javascript @dev @destructive
+  Scenario: Add Landing Page
```
Comment (cjwest) [inline products/jse_sub/features/jse_site_owner.feature:37]: This duplicates the tests in this file:
SU-SWS/permissions_tests/create_delete_landing_page/create_delete_landing_page.feature

Let's keep them in one place.

File: products/jse_sub/features/jse_site_owner.feature [hunk 13]
```diff
@@ -0,0 +1,124 @@
+@api
```
Comment (cjwest) [inline products/jse_sub/features/jse_site_owner.feature]: What's the "jse_sub" directory?

File: products/jse_sub/behat.yml [hunk 14]
```diff
@@ -0,0 +1,15 @@
+imports:
```
Comment (cjwest) [inline products/jse_sub/behat.yml:1]: Why do we have a jse_sub directory?

File: includes/features/SU-SWS/permissions_tests/edit_main_menu/edit_main_menu.feature [hunk 15]
```diff
@@ -0,0 +1,24 @@
+Feature: Ensure Site Owners and Editors can access the edit main menu page
```
Comment (cjwest) [inline includes/features/SU-SWS/permissions_tests/edit_main_menu/edit_main_menu.feature:1]: Update description "Editors cannot access...."

File: includes/features/SU-SWS/permissions_tests/create_delete_news/create_delete_news.feature [hunk 16]
```diff
@@ -0,0 +1,98 @@
...
+  Then I should see "News Item Bar has been updated" in the "Console" region
+  And I should be on "news/bar"
+  And I am on "news/bar"
+  And I click "Edit" in the "Content Head" region
+  And I click on the element with css selector "#edit-delete"
+  And I click on the element with css selector "#edit-submit"
+  Then I should see the text "News Item Bar has been deleted" in the "Console" region
+
+  @api @dev @destructive @javascript
+  Scenario: Create news item as site owner, edit as editor, delete as site owner
```
Comment (cjwest) [inline includes/features/SU-SWS/permissions_tests/create_delete_news/create_delete_news.feature:29]: I'm not sure why we need this. Editors can create news items, or at least they should be able to.

File: includes/features/SU-SWS/permissions_tests/create_delete_news/create_delete_news.feature [hunk 17]
```diff
@@ -0,0 +1,98 @@
+Feature: Ensure Site Owners can create and delete news items
+  In order to ensure that users with permissions can create, edit and delete news items
+  As a user with site permissions
+  I want to be able create, edit and delete news items
+
+  @api @dev @destructive @javascript
+  Scenario: Create, edit, then delete news item as site owner
```
Comment (cjwest) [inline includes/features/SU-SWS/permissions_tests/create_delete_news/create_delete_news.feature:7]: What about including this in the "jse_site_owner.feature" file?

File: includes/features/SU-SWS/permissions_tests/create_delete_publications/create_delete_publications.feature [hunk 18]
```diff
@@ -0,0 +1,98 @@
...
+  And I should be on "publications/bar"
+  Given I am logged in as a user with the "site owner" role
+  And I am on "publications/bar"
+  And I click "Edit" in the "Content Head" region
+  And I click on the element with css selector "#edit-delete"
+  And I click on the element with css selector "#edit-submit"
+  Then I should see the text "Publication Bar has been deleted" in the "Console" region
+
+  @api @dev @destructive @javascript
+  Scenario: Create publication as editor, edit and delete as a site owner
```
Comment (cjwest) [inline includes/features/SU-SWS/permissions_tests/create_delete_publications/create_delete_publications.feature:77]: Do we need all these scenarios?

File: includes/features/SU-SWS/permissions_tests/edit_all_pages/edit_all_pages.feature [hunk 19]
```diff
@@ -0,0 +1,29 @@
+Feature: Ensure Site Owners and Editors can access manage all pages
+  In order to ensure that site owners can access manage all pages
```
Comment (cjwest) [inline includes/features/SU-SWS/permissions_tests/edit_all_pages/edit_all_pages.feature:2]: How about we move this into jse_site_owner.feature and jse_editor.features

**Comment by cjwest** [comment]
@kmakitan, what's the status of this branch?

**Comment by kmakitan** [comment]
@cjwest We can delete this one, all the changes are in the create_delete_event branch

**Pull request closed without merging**